### ISSUE TITLE
`explicit-length-check`: ignore variables with non-array properties?

### ISSUE THREAD

**Issue by bmish** [sindresorhus/eslint-plugin-unicorn#1266]
```js
if (
  packagingData.dimensions.width &&
  packagingData.dimensions.height &&
  packagingData.dimensions.length // violation reported by unicorn/explicit-length-check
) { }
```

In this example, `packagingData.dimensions` is an object, and `length` could be a string. 

As a heuristic to avoid false positives with detecting `length` or `size` on variables that are not actually arrays, could we ignore variables that have non-array properties? For example, an array does not have a `width` property, so if we see that used, we would assume `dimensions` is not an array, and ignore it.

**Comment by fisker** [comment]
> For example, an array does not have a width property, so if we see that used, we would assume dimensions is not an array, and ignore it.

I like the idea.
